udfs/CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+# Contributing UDFs
+
+Thank you for taking the time to contribute to this repository of user-defined
+functions (UDFs) for BigQuery.
+
+The following is a set of guidelines for contributing a UDF to this repository.
+
+## UDF Contribution Guidelines
+
+### Add your UDF
+
+1.  Add your UDFs (**one** UDF per file) and a
+    [Contributor License Agreement](#contributor-license-agreement) to the
+    appropriate directory.
+    *   If your function replicates logic from some other data warehouse UDF,
+        place it in the relevant sub-directory in the
+        [migration](/udfs/migration) directory. Otherwise, place it in the
+        [community](/udfs/community) directory.
+1.  Add test cases for your UDFs.
+    *   Edit the `test_cases.yaml` file to include test inputs and expected
+        outputs for the function.
+    *   Make sure test cases provide full coverage of the function's expected
+        behavior. For example, if integers are the expected input, please
+        provide test cases with the following inputs: negative numbers, zero,
+        positive numbers, and null values.
+1.  Describe what your UDF does.
+    *   Edit the `README.md` in the associated sub-directory to include a
+        description of the function and make sure your function is listed in
+        alphabetical order amongst the other functions in the `README.md`.
+    *   Make sure the same description is placed as a comment in your UDF file.
```

Comment:
nit / nice to have in future PR:
1) alphabetizing is annoying and error prone (speaking for dyslexic experience)
2) Keeping comments / README in sync is also annoying / error prone.

Solution:
DRY up documentation by auto-generate these READMEs instead of having this burden on contributor.
Much like javadoc or sphinx or terraform-docs These udf  READMEs should generated from the the comment in the UDF SQL files.